### ISSUE TITLE
Actions trigger the more-info window

### ISSUE THREAD

**Issue by mgiako** [benct/lovelace-multiple-entity-row#217]
Hi,

I have this entity:

![image](https://user-images.githubusercontent.com/44270368/145708696-5f650459-8f9c-4a31-ab65-efe35d5a1342.png)

which checks the status and can restart an ESP32 on esphome

the code is:
```
              - entity: binary_sensor.trackerterrazzaconnected
                show_state: false
                icon: mdi:car-esp
                type: custom:multiple-entity-row
                name: Terrazza
                entities:
                  - entity: sensor.trackerterrazza_signal
                    state_color: true
                    name: RSSI

                  - entity: sensor.trackerterrazza_uptime
                    state_color: true
                    name: uptime
                    format: total

                  - entity: binary_sensor.trackerterrazzaconnected
                    icon: mdi:wifi
                    state_color: true
                    name: false
                    tap_action:
                      action: call-service
                      service: switch.turn_on
                      service_data:
                        entity_id: switch.trackerterrazzarestart
                      confirmation:
                        text: "Restart ESP Terrazza?"
```

Since 2021.12.b0 (and still now on 2021.12.0) when I click on the wifi icon the action is performed but **it also opens the more-info window of the main element**: binary_sensor.trackerterrazzaconnected. 
The window opens even if I click cancel in the confirmation.

I have other similar entities, and it happens on all of them, except on those that have an input text as their main entity.

It did not happen with HA 2021.11.5.

Thanks

**Comment by bkbilly** [comment]
I have exactly the same problem, so I've looked for other options.

My temporary solution works only on desktop and it disables the more-info of the main element, but you can use the entities normally.
```
tap_action: none
hold_action: none
double_tap_action: none
```

Hope a more permanent solution is found.

**Comment by GuGu927** [comment]
Here's same problem.
Click entity at "entities" section dose not work.
Just main entity is poped up.

**Comment by GuGu927** [comment]
@bkbilly Thanks. Can use entities, but not entity.

**Comment by bkbilly** [comment]
Yeah, I couldn't find a solution for that. It disables all events of the main element `binary_sensor.trackerterrazzaconnected`.

**Comment by gumileeev** [comment]
Same problem. When you click on an additional element, the main element window opens. When clicked, all elements of the group are selected.

**Comment by NickM-27** [comment]
I am having this issue as well, not sure if it is an issue for this plugin to solve or an issue for HA to solve?

**Comment by ildar170975** [comment]
The plugin was not changed.
HA changed.
Probably there are some general changes regarding handling actions.

**Comment by NickM-27** [comment]
Right, but it could be a bug introduced by HA update or it could be HA updated the requirements and the plug-in needs to accommodate that new change

**Comment by ildar170975** [comment]
Absolutely agree.
Let's wait for the plugin's author.
Hope he will help us.

**Comment by NickM-27** [comment]
I might be able to submit a PR if I find time tomorrow after work. I took a look tonight and I see the plug-in relies on `custom-card-helper` for clickHelper so it's possible that's where the issue lies. Can't tell if that comes from HA or somewhere else yet.

**Comment by NickM-27** [comment]
Looks like that is a dependency that comes from here: https://github.com/custom-cards/custom-card-helpers

**Comment by TobiasSchwarz74** [comment]
Hi,
i have the same issue, but found out, that as long, as the main entity is a switch or light domain, everything works fine.
As soon, as the main entity is any other domain (i.e. a binary_sensor), tap actions on entities no longer working on touch devices.
On a desktop everything works fine with a mouse.

Example:
The first row uses a switch as main entity - tap_action works fine.
The second row uses a binary_sensor (i.e. a template sensor to customize the icon and state depending on the state of a different entity) - tap_action does not work since 2021.12.

```
type: entities
entities:
  - type: custom:multiple-entity-row
    entity: switch.switch_1
    show_state: false
    tap_action: none
    entities:
      - entity: switch.switch_1
        tap_action:
          action: toggle
  - type: custom:multiple-entity-row
    entity: binary_sensor.binary_sensor_1
    show_state: false
    tap_action: none
    entities:
      - entity: switch.switch_1
        tap_action:
          action: toggle
```

**Comment by NickM-27** [comment]
@TobiasSchwarz74 Wow yeah I see the same thing, that is very odd but a great clue

**Comment by TobiasSchwarz74** [comment]
> @TobiasSchwarz74 Wow yeah I see the same thing, that is very odd but a great clue

took me half of the day to find out in which situations this happens ;-)

**Comment by NickM-27** [comment]
I found something else interesting too. When I have the click action as more-info and I click on a sensor different to the main entity, the title of the more-info popup is the same as the main entity however the graph actually shows the correct information
![wrong](https://user-images.githubusercontent.com/14866235/146070
![sorta_right](https://user-images.githubusercontent.com/14866235/146070201-45f64363-fbbb-4254-9069-b75e80539bd1.png)
189-33d626d5-3923-4b8e-85b3-2d65e8a195d6.png)
.

**Comment by Arman-GeDon** [comment]
> I found something else interesting too. When I have the click action as more-info and I click on a sensor different to the main entity, the title of the more-info popup is the same as the main entity however the graph actually shows the correct information
> 
If you wait a few seconds it will switch the graph to the main entity

**Comment by chengjwk** [comment]
It appears to be working for me now with 2021.12.2

**Comment by bkbilly** [comment]
I've also updated on the new version `2021.12.2` and I've reloaded lovelace, but I still can't get the more-info appear on the other entities.

**Comment by NickM-27** [comment]
I also updated and am still seeing the issue

**Comment by Arman-GeDon** [comment]
Of course updating HA will fix nothing. The card needs to be adjusted to HA changes in release 2021.12.x

**Comment by chengjwk** [comment]
I'm not sure what to say. I've upgraded and the card is back to working they way it was prior to this bug appearing.

**Comment by TobiasSchwarz74** [comment]
No, still not working with 2021.12.2

**Comment by benct** [comment]
HA added a new [action handler on the generic-entity-row](https://github.com/home-assistant/frontend/commit/cf062bf0f4395c4c474fa571aba5add774589d28#diff-0d79a5af52b883024274a9139813d2646eea9ab711374a5fc9d92228f141acefR147) which (intentionally or not) intercepts clicks on any additional entities added with multiple-entity-row. It still works as normal with some entity domains (like switch or light) because they disable this action handler for these specific domains.

Working on finding a workaround for this..

**Comment by TobiasSchwarz74** [comment]
Very interesting, why this changed behavior is only visible on touch devices and can't be seen on a desktop controlled with a mouse?!

**Comment by Arman-GeDon** [comment]
It's visible on a desktop controlled with a mouse

**Comment by benct** [comment]
Should (hopefully) be fixed in version `4.3.0`.

**Comment by NickM-27** [comment]
@benct So it is working great on desktop browser, but in the mobile app it seems to still be incorrect

**Comment by mgiako** [comment]
Same here: ok on browser, no ok on mobile.

**Comment by Arman-GeDon** [comment]
@benct 
v4.3.0
It does not work on mobile (cleared app cache)
It does not work in browser (newest Chrome and Opera, cleared cache) in one scenario: the main entity is a switch/light with toggle: true and the rest are sensors (entities or attributes). For example:

`  - entity: switch.cooker
    type: custom:multiple-entity-row
    secondary_info: last-changed
    toggle: true
    state_color: true
    entities:
      - attribute: voltage
        unit: V
      - attribute: current_consumption
        unit: W
`

**Comment by NickM-27** [comment]
@Arman-GeDon I would make sure lovelace is reloaded, I have this exact scenario and it works fine in the browser

**Comment by Arman-GeDon** [comment]
@NickM-27 
On mobile, on a browser? Cache cleared, lovelace reloaded, the app/browser killed/closed, even HA restarted (which is unecessary)

**Comment by NickM-27** [comment]
> @NickM-27 
> 
> On mobile, on a browser? Cache cleared, lovelace reloaded, the app/browser killed/closed, even HA restarted (which is unecessary)

I don't think your comment made it clear you were using a mobile browser. Yes, using a mobile browser it also doesn't work. But desktop browser does work.

**Comment by Arman-GeDon** [comment]
@NickM-27 
My comment is clear :)
Mobile app on mobile phone (android 10)
desktop browser on PC (windows 7)

**Comment by NickM-27** [comment]
It's not clear, I mentioned it working in browser and you brought up mobile browser, but now mention desktop browser. So is it or is it not working correctly for you on the desktop browser?

**Comment by Arman-GeDon** [comment]
@NickM-27 
I have never mentioned mobile browser.
I said it's not working on mobile app at all and it's not working on desktop browser with secondary entities in specific case. 

Once again:
It does not work on mobile (cleared app cache), killed app, restarted HA.
It does not work in DESKTOP browser (newest Chrome and Opera, cleared cache, lovelace reloaded, browser closed and open) in one scenario: the main entity is a switch/light with toggle: true and the rest are sensors (entities or attributes). For example:
```

  - entity: switch.cooker
    type: custom:multiple-entity-row
    secondary_info: last-changed
    toggle: true
    state_color: true
    entities:
      - attribute: voltage
        unit: V
      - attribute: current_consumption
        unit: W

```

**Comment by tomlut** [comment]
Latest update fixed this for me. Thank you!

**Comment by NickM-27** [comment]
@Arman-GeDon Your code snippet does not show specific tap actions set for the entities, at this point I'm not even sure what you are saying doesn't work

**Comment by ildar170975** [comment]
Desktop, Win10x64, Chrome:
works fine

iPad Air 2 (iOS 15.x):
tapping on secondary entities leads to more-info for the main entity

iPhone 5S (iOS 12.x):
tapping on secondary entities :
- either leads to more-info for the this entity
- or does nothing

**Comment by Arman-GeDon** [comment]
@NickM-27 
My bad, you're right. I was wrongly using attributes and not entities for sensors, which of course leads to main entity (attributes don't have history). 
Desktop browser works

**Comment by kineticscreen** [comment]
Yep fixed on desktop, but not only does it still open up more-info on the Android app, but it doesn't also do the tap action.

**Comment by matze338** [comment]
Still Not Work on configurations with climate entities 😕
````
entity: climate.wohnzimmer_rechts
type: custom:multiple-entity-row
show_state: true
attribute: temperature
unit: °C
styles:
  width: 50px
entities:
  - entity: sensor.wohnzimmer
    name: false
    styles:
      width: 50px
  - entity: sensor.wohnzimmer_2
    name: false
    styles:
      width: 50px

**Comment by kineticscreen** [comment]
Oh yeah sorry mine was with a climate entity too.

**Comment by guidokieboom** [comment]
Does not yet work with 4.3.0 on mobile device (iOS).
Works different with 4.3.0 on desktop (Safari). Attribute result is now different. See screenshots (first one is from iOS; second is from desktop-Safari)

<img width="432" alt="Screenshot 2021-12-18 at 07 48 41" src="https://user-images.githubusercontent.com/65387179/146632455-bee9f606-d12d-46b3-bdb9-c46bb1ae16c7.png">

<img width="982" alt="Screenshot 2021-12-18 at 07 43 01" src="https://user-images.githubusercontent.com/65387179/146632458-69f01392-063b-4900-bc43-a2652adf1833.png">

**Comment by j9brown** [comment]
Thanks for looking into this issue.  I noticed that the [paper-buttons-row plugin](https://github.com/jcwillox/lovelace-paper-buttons-row) which is quite similar to multiple-entity-row seems to be immune to the problem.  Perhaps we could learn something from how it wires up its action handlers.

**Comment by benct** [comment]
> Attribute result is now different. See screenshots (first one is from iOS; second is from desktop-Safari)

@guidokieboom Could you create a separate issue? That seems not directly related to this issue.

> I noticed that the [paper-buttons-row plugin](https://github.com/jcwillox/lovelace-paper-buttons-row) which is quite similar to multiple-entity-row seems to be immune to the problem. Perhaps we could learn something from how it wires up its action handlers.

That card appends the content to a new row, and does so in a completely differently way than here, unfortunately.

**Comment by benct** [comment]
There will likely come a fix for this issue from HA, soon. See [#10961](https://github.com/home-assistant/frontend/issues/10961#issuecomment-997370789) for more info.

**Comment by NickM-27** [comment]
Awesome, glad to see it.

**Comment by NickM-27** [comment]
@benct I see that the frontend fix mentioned made it into `2021.12.4` today. I updated but unfortunately am still seeing this issue on mobile, not sure if it still requires a fix on the this integrations side?

Thanks!

**Comment by benct** [comment]
> ... not sure if it still requires a fix on the this integrations side?

It does. Should be fixed in version `4.3.2` (clear cache, etc..).

**Comment by Arman-GeDon** [comment]
@benct Seems working on mobile after clearing app cach. Big thanks!

**Comment by Arman-GeDon** [comment]
@benct 
What is not working is more-info on hold/double tap action when different than main entity is specified (desktop, mobile).
For example I'm expecting more-info on hold action for sensor.sensor3 but get for main entity (sensor.sensor1)
```
  - entity: sensor.sensor1
    type: custom:multiple-entity-row
    name: Energy
    secondary_info: last-updated
    show_state: false
    hold_action:
      action: more-info
      entity: sensor.sensor3
    entities:
      - entity: sensor.sensor2
        name: false
```

**Comment by NickM-27** [comment]
@benct works great! Thank you!

**Comment by NickM-27** [comment]
@Arman-GeDon as per the README, those are not supported for extra entities. Only tap_action is supported. So that's a feature request (I'd sure like to see too 👌) but not apart of this bug.

**Comment by Arman-GeDon** [comment]
@NickM-27 
As you can see I'm using hold action for main entity and overriding entity-id as per the README

**Comment by benct** [comment]
> What is not working is more-info on hold/double tap action when different than main entity is specified (desktop, mobile).

See #188.

Closing issue, thanks for all feedback 🙂

**Comment by guidokieboom** [comment]
works for me now (no more info popup). Thanks!!
attribute results no longer different… so no separate issue created.

**Comment by herr2d2** [comment]
@benct  Works perfect ... Thanks a lot!

**Comment by afarago** [comment]
Many thanks for your efforts, works fine now!

**Comment by poohbah** [comment]
Confirmed working again on android mobile but ONLY after clearing the app cache.

**Comment by mattthewclayton** [comment]
Sorry for reviving a closed issue, but I don't want to create a new one if I don't need to and I am just doing something stupid. My card works as expected on desktop, but not on iPhone. 

In order for the toggle action to work on the entity (not the row) I have to do a long (3D?) press, and even then it's fairly intermittent if it works, you have to be very precise with your finger it seems.

Is this likely an issue with this integration (and I will open a new issue), or with the iOS companion app?

HA Version: 2022.5.0
Companion App: 2022.3

<img width="473" alt="Screenshot 2022-05-14 at 12 02 21" src="https://user-images.githubusercontent.com/9895197/168423332-18729f5e-f3fb-4588-aa17-e56679b9a962.png">

```type: entities
entities:
  - entity: climate.living_room
    type: custom:multiple-entity-row
    icon: mdi:temperature-celsius
    secondary_info:
      attribute: current_temperature
      unit: °C
    name: Living
    tap_action: none
    show_state: false
    entities:
      - entity: light.living_room
        icon: mdi:ceiling-light
        name: Living
        tap_action:
          action: toggle
      - entity: light.dining_room
        icon: mdi:ceiling-light
        name: Dining
        tap_action:
          action: toggle
      - entity: light.kitchen
        icon: mdi:ceiling-light
        name: Kitchen
        tap_action:
          action: toggle
      - entity: switch.left_lamp
        icon: mdi:lamp
        name: Left
        tap_action:
          action: toggle
      - entity: switch.right_lamp
        icon: mdi:lamp
        name: Right
        tap_action:
          action: toggle
```

**Comment by tomlut** [comment]
Please don;t comment on closed (and unrelated) issues. Open a new one.
